ntbs-service/Pages/Notifications/Edit/MDRDetails.cshtml.cs

```diff
@@ -0,0 +1,127 @@
+            MDRDetails = Notification.MDRDetails;
+            await SetNotificationProperties(isBeingSubmitted, MDRDetails);
+
+            if (MDRDetails.ShouldValidateFull)
+            {
+                TryValidateModel(MDRDetails, MDRDetails.GetType().Name);
+            }
+
+            return Page();
+        }
+
+        protected override IActionResult RedirectToNextPage(int notificationId, bool isBeingSubmitted)
+        {
+            // This is the last page in the flow, so there's no next page to go to
+            return RedirectToPage("./MDRDetails", new { id = notificationId, isBeingSubmitted });
+        }
+
+        protected override async Task ValidateAndSave()
+        {
+            UpdateFlags();
+            await ValidateRelatedNotification();
+            MDRDetails.SetFullValidation(Notification.NotificationStatus);
+
+            if (TryValidateModel(MDRDetails.GetType().Name))
+            {
+                await Service.UpdateMDRDetailsAsync(Notification, MDRDetails);
+            }
+        }
+
+        private async Task ValidateRelatedNotification()
+        {
+            if (MDRDetails.RelatedNotificationId != null) {
+                var relatedNotification = await GetRelatedNotification(MDRDetails.RelatedNotificationId.Value);
+                if (relatedNotification == null)
+                {
+                    ModelState.AddModelError("MDRDetails.RelatedNotificationId", ValidationMessages.RelatedNotificationIdInvalid);
+                }
+            }
+        }
+
+        private async Task<Notification> GetRelatedNotification(int noticicationId)
+        {
+            return await NotificationRepository.GetNotificationAsync(noticicationId);
+        }
+
+        private void UpdateFlags()
+        {
+            if (MDRDetails.ExposureToKnownCaseStatus != Status.Yes || MDRDetails.CaseInUKStatus != Status.Yes)
+            {
+                MDRDetails.RelatedNotificationId = null;
+                ModelState.Remove("MDRDetails.RelatedNotificationId");
+            }
+            if (MDRDetails.ExposureToKnownCaseStatus != Status.Yes || MDRDetails.CaseInUKStatus != Status.No)
+            {
+                MDRDetails.CountryId = null;
+                ModelState.Remove("MDRDetails.CountryId");
+            }
+
+            if (MDRDetails.ExposureToKnownCaseStatus != Status.Yes)
+            {
+                MDRDetails.RelationshipToCase = null;
+                MDRDetails.CaseInUKStatus = null;
+                ModelState.Remove("MDRDetails.RelationshipToCase");
+                ModelState.Remove("MDRDetails.CaseInUKStatus");
+            }
+        }
+
+        public ContentResult OnGetValidateMDRDetailsProperty(string key, string value, bool shouldValidateFull)
+        {
+            return ValidationService.ValidateModelProperty<MDRDetails>(key, value, shouldValidateFull);
+        }
+
+        public async Task<ContentResult> OnGetValidateMDRDetailsRelatedNotificationAsync(string value)

```

Comment:
What happens if this method only accepts int and you hit it with a non-int value? Does it just 404?

Reply:
I remember trying to use int values in these validation methods and running into issues when razor was trying to map them - so I'm not sure

Reply:
definitely worth trying to use `int` and if there are problems leaving a comment that we're expecting int here but can't say so because of xyz

Reply:
@jasiekmiko Just tried changing to int, what happens is as follows:  Any input containing text is parsed to 0 serverside, so I could remove all the stuff about typechecking and it would just return the error message saying this ID wasn't found. Shall i do this, or stick with explicit checks for integer ids?

Reply:
hmm, I think that seems nicer. is there a chance that using `int?` make it any more intuitive?

Reply:
I'm assuming we're providing a numbers-only box on the front-end, so they'd have to hack-around to see that message anyways, so it's fine for the message not be simple and not super polished.

Reply:
@jasiekmiko you can put "e"s into a numbers only box for powers. Ok, I'll just remove the type check and go for the normal error message as it doesn't make a big difference to the user, but simplifies the code a lot.

Reply:
@jasiekmiko Actually that doesn't work because it to int? means that non-string entries are parsed to null, and not validated (because actual null, as in no entry, is valid) and just int means we can't capture empty, valid entry. I think I just need to leave it as it is for this field and check explicitly for integer input...

Reply:
ah that makes sense. yeah `int` sounds like the right approach then! :)